### ISSUE TITLE
Changing from v0.2.6 to v0.2.7 in same conda environment causes numpy error

### ISSUE THREAD

**Issue by matscorse** [icenet-ai/icenet#214]
* IceNet version: v0.2.6 -> v0.2.7
* Operating System: BAS HPC

### Description

Looking through requirements history up to v0.2.6, numpy wasn't explicitly included. It should be pinned to <1.24 at the latest for v0.2.6 (when h5py was pinned to 2.10.0).
With numpy v1.21-1.23 would've just given a deprecation warning. And, older versions of numpy would work fine.
So, get following error if install v0.2.6 or install v0.2.6, then try to install v0.2.7_dev on top of it:

> ```
>   File "h5py/h5t.pxd", line 14, in init h5py._conv
>   File "h5py/h5t.pyx", line 293, in init h5py.h5t
>   File "/data/hpcdata/users/bryald/miniconda3/envs/icenet_dit_test_matt/lib/python3.8/site-packages/numpy/__init__.py", line 320, in __getattr__
>     raise AttributeError("module {!r} has no attribute "
> AttributeError: module 'numpy' has no attribute 'typeDict'
> ```

### What I Did

```
temporary solution was
> conda install -c conda-forge h5py
inside the conda environment.
Otherwise a 'fresh' conda environment for v0.2.7 solves this error.
```

**Comment by bnubald** [comment]
Thanks for posting the issue Matt.

**For posterity:**

Issue was relating to a dependency specification.

IceNet v0.2.6 and prior pinned `h5py` to `2.10.0` which requires numpy `<=1.21`, I believe up 1.23 (inclusive) would also work, but just with deprecation warnings.

Since numpy wasn't pinned, the environment would install `numpy` v1.24 (latest supported for Python 3.8). Due to typeDict (and other methods being removed), the older pinned h5py version would cause the error mentioned in this issue.

Running `conda install -c conda-forge h5py` above cleared the issue - because it installed the newer version of h5py which is compatible with numpy v1.24.

This relates to #87 since to do with similar deprecation issue with numpy v1.24.

The fix mentioned there of installing `numpy` v1.23.5 would also work here as an alternate (though, likely with deprecation warnings).

**Comment by bnubald** [comment]
Going from IceNet v0.2.6 or prior to v0.2.7+ as it stands would still cause this error since h5py has been unpinned, but not forced to be updated.

Might be worth setting h5py version >2.10.0, so pip will update if older icenet installed, so that it updates properly.
